File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    // -ve number cannot be a plaindrome
    if(x < 0) return false;

    const s = String(x);
    let i=0;
    let j = s.length - 1;
    while(i < j) {
        if(s[i] !== s[j]) return false;
        i++;
        j--;
    }

    return true;
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # negative numbers are NOT palindrom, as given
        if x < 0:
            return False
        
        # let's break the number down into an array
        arr = []
        while x > 0:
            arr.append(x%10)
            x = x // 10
        return self.isArrayPalindrome(arr)


    def isArrayPalindrome(self, arr: list) -> bool:
        i = 0
        j = len(arr) - 1
        while i < j:
            if arr[i] != arr[j]:
                return False
            i += 1
            j -= 1
        return True

        
        